### PR TITLE
motion: refactor manager to use flags and remove motion-controller

### LINKED ISSUE #151: movement controller: remove cmd and build folders
The top level cmd and build folders are only used for testing the movement controller, which makes the project structure confusing.

# Option 1: simple merge to other folders

Something like:
- [ ] Remove `Dockerfile.movement-controller` that is a duplicate of `manager/Dockerfile` 
- [ ] Move `Dockerfile.dummy-mover` to `test`
- [ ] Integrate Makefile with `manager/Makefile` and `Makefile` while removing duplications (only needs `movement-controller` target that go builds it and a target for dummy mover)
- [ ] Remove the `build` folder
- [ ] Move `movement-controller/main.go` to `manager` or a subfolder within
- [ ] Remove `cmd` folder

# Option 2: optional controllers using program arguments

Change main.go in manager to only start controllers (and related logic) if a flag is on. This way the same binary can be used to run just the movement controller. The dummy should still probably be part of test. Overlay for deploying just movement is needed. 

# Option 3: Plan for a future usage of build and cmd

Keep as is but create a readme that explains why it's there. Probably need to use just one way to run movement controller and if it's a separate binary then stick to it.

### PR DESCRIPTION
**Pull request by froesef**
Fixes #151

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -50,32 +50,34 @@ undeploy:
...
 	$(MAKE) -C test/services docker
 
 .PHONY: docker-build
 docker-build:
-	$(MAKE) -C build docker-build-all
+	$(MAKE) -C test/dummy-mover docker-all
 	$(MAKE) -C manager docker-build
 	$(MAKE) -C secret-provider docker-build
 	$(MAKE) -C connectors docker-build
+	$(MAKE) -C test/dummy-mover docker-build-dummy-mover
```
Comment (roee88) [inline Makefile:72]: Where is this target drfined?

File: Makefile [hunk 2]
```diff
@@ -50,32 +50,34 @@ undeploy:
...
+	$(MAKE) -C test/dummy-mover docker-all
 
 # Build only the docker images needed for integration testing
 .PHONY: docker-minimal-it
 docker-minimal-it:
-	$(MAKE) -C build docker-dummy-mover
 	$(MAKE) -C manager docker-all
 	$(MAKE) -C secret-provider docker-all
+	$(MAKE) -C test/dummy-mover docker-all
 	$(MAKE) -C test/services docker
```
Comment (shlomitk1) [inline Makefile:64]: Why others get docker-all and services - only docker?

File: Makefile [hunk 3]
```diff
@@ -50,32 +50,34 @@ undeploy:
...
-	$(MAKE) -C build docker-build-all
+	$(MAKE) -C test/dummy-mover docker-all
 	$(MAKE) -C manager docker-build
 	$(MAKE) -C secret-provider docker-build
 	$(MAKE) -C connectors docker-build
+	$(MAKE) -C test/dummy-mover docker-build-dummy-mover
 
 .PHONY: docker-push
 docker-push:
 	$(MAKE) -C build docker-push-all
```
Comment (shlomitk1) [inline Makefile:76]: I don't understand the decision when "-all" is used. Make -C build docker-build-all is not used but docker-push-all is. How does this work?

File: Makefile (hunk 1 above)
Comment (froesef) [reply Makefile]: Sorry had the wrong target in there after refactoring.

File: Makefile (hunk 3 above)
Comment (froesef) [reply Makefile]: I unified the target names. Hope it's fine now.

**Comment by roee88** [comment]
@froesef please rebase and resolve conflicts, then this could be approved.

**Review by shlomitk1 — APPROVED** [review]

**Pull request merged**